### PR TITLE
feat: add channel alerts in send and receive pages

### LINKED ISSUE #241: Add alerts on wallet send/receive pages if receiving/sending capacity is insufficient
Similar to the Node page (see the alerts there)

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #241 

What if the user doesn't have any channels? Should we hide the Send/Receive buttons from the Wallet screen

Screenshot 2024-07-19 at 12 43 33 PM

## Send
Screenshot 2024-07-19 at 12 44 14 PM

## Receive
Screenshot 2024-07-19 at 12 43 59 PM

### CODE REVIEW

File: frontend/src/screens/wallet/Send.tsx [hunk 1]
```diff
@@ -106,6 +119,27 @@ export default function Send() {
...
+        <>
+          {/* If all channels have less or equal balance than their reserve, show a warning */}
+          {channels?.every(
+            (channel) =>
+              channel.localBalance <=
+              channel.unspendablePunishmentReserve * 1000
+          ) && (
+            <Alert>
+              <AlertTriangle className="h-4 w-4" />
+              <AlertTitle>Channel reserves unmet</AlertTitle>
```
Comment (bumi) [inline frontend/src/screens/wallet/Send.tsx:132]: Shouldn't this be something like: 

You don't have enough spending balance?

File: frontend/src/screens/wallet/Receive.tsx [hunk 2]
```diff
@@ -116,6 +127,27 @@ export default function Receive() {
         title="Receive"
         description="Create a lightning invoice that can be paid by any bitcoin lightning wallet"
       />
+      {!!channels?.length && (
+        <>
+          {/* If all channels have less than 20% incoming capacity, show a warning */}
+          {channels?.every(
+            (channel) =>
+              channel.remoteBalance <
+              (channel.localBalance + channel.remoteBalance) * 0.2
```
Comment (bumi) [inline frontend/src/screens/wallet/Receive.tsx:136]: can this also check the amount? that if I enter a high amount it will also show the message?

Comment (im-adithya) [reply frontend/src/screens/wallet/Receive.tsx:136]: As in if amount is > than receiving capacity of every channel / we consider MPP and check with net receiving capacity?

Or we can also show an error if the amount entered is > receiving capacity (But is it good to restrict the user like that?)

File: frontend/src/screens/wallet/Send.tsx (hunk 1 above)
Comment (im-adithya) [reply frontend/src/screens/wallet/Send.tsx]: Changed to "Low spending balance"

File: frontend/src/screens/wallet/Receive.tsx (hunk 2 above)
Comment (im-adithya) [reply frontend/src/screens/wallet/Receive.tsx]: Done!

File: frontend/src/screens/wallet/Receive.tsx [hunk 3]
```diff
@@ -116,6 +127,28 @@ export default function Receive() {
         title="Receive"
         description="Create a lightning invoice that can be paid by any bitcoin lightning wallet"
       />
+      {!!channels?.length && (
```
Comment (rolznz) [inline frontend/src/screens/wallet/Receive.tsx:130]: need to update this check to also only happen if the backend type has channel management (the useInfo hook exposes this, see other places in the source code)

File: frontend/src/screens/wallet/Receive.tsx [hunk 4]
```diff
@@ -116,6 +127,28 @@ export default function Receive() {
...
+        <>
+          {/* If all channels have less than 20% incoming capacity, show a warning */}
+          {channels?.every(
+            (channel) =>
+              channel.remoteBalance <
+                (channel.localBalance + channel.remoteBalance) * 0.2 ||
```
Comment (rolznz) [inline frontend/src/screens/wallet/Receive.tsx:136]: I don't think we need to loop through channels, can we use the balances instead? (see `useBalances`)

File: frontend/src/screens/wallet/Receive.tsx (hunk 3 above)
Comment (rolznz) [reply frontend/src/screens/wallet/Receive.tsx:130]: (also applies to send)

File: frontend/src/screens/wallet/Receive.tsx (hunk 4 above)
Comment (rolznz) [reply frontend/src/screens/wallet/Receive.tsx:136]: (also applies to send)

**Comment by rolznz** [comment]
@im-adithya 

> What if the user doesn't have any channels? Should we hide the Send/Receive buttons from the Wallet screen

Good question, I think we should have a specific alert for this on the main wallet page, and not allow the user to enter the send or receive pages (If the LNClient has channel management and the user has no channels)

File: frontend/src/screens/wallet/Receive.tsx (hunk 4 above)
Comment (im-adithya) [reply frontend/src/screens/wallet/Receive.tsx]: Done!

**Review by rolznz — APPROVED** [review]
tACK

**Pull request merged**